src/geometry.cpp

```diff
@@ -536,6 +536,196 @@ extern "C" double openmc_calculate_optical_thickness(
   return optical_thickness;
 }
 
+extern "C" void openmc_get_voxel_cross_sections(Position voxel_min,
+  Position voxel_max, int num_samples, double* sigma_t_output,
+  double* sigma_a_output)
+{
+  double total_sigma_t = 0.0;
+  double total_sigma_a = 0.0;
+  SpatialBox voxel_box(voxel_min, voxel_max);
+  int64_t id = 1;
+  uint64_t seed = init_seed(id, STREAM_SOURCE);
+  int valid_samples = 0;
+
+#pragma omp parallel for reduction(                                            \
+    + : total_sigma_t, total_sigma_a, valid_samples)
+  for (int i = 0; i < num_samples; ++i) {
+    Position sampled_position = voxel_box.sample(&seed);
+
+    Particle p;
+    SourceSite site;
+    site.E = 1.0;
+    site.particle = ParticleType::neutron;
+    site.r = sampled_position;
```

Comment:
[readability] There is no reason to make a separate variable `sampled_position` since you only use it once:
```suggestion
    site.r = voxel_box.sample(&seed);
```